samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -143,11 +143,11 @@ public synchronized String getEphemeralPath() {
    * @return processor's data
    */
   String readProcessorData(String fullPath) {
-    String data = zkClient.<String>readData(fullPath, true);
-    if (data == null) {
-      throw new SamzaException(String.format("Cannot read ZK node:", fullPath));
+    try {
+      return zkClient.readData(fullPath, false);
```

Comment:
Will make the change.
Possible exceptions:
A) NoNodeExistsException(Thrown when zknode doesn't exist).
B) ZkConnectTimeout, ZkSessionTimeout (connection timeout exception are possible). 

When zookeeper wasn't properly setup, experienced B) in tests.